### ISSUE TITLE
Missing API documentation for Intuit.Ipp.OAuth2PlatformClient.OAuth2Client

### ISSUE THREAD

**Issue by pideltajah** [intuit/QuickBooks-V3-DotNET-SDK#178]
There seems to be a hole in the [API reference documentation](https://static.developer.intuit.com/sdkdocs/qbv3doc/ippdotnetdevkitv3/html/ccf8123c-8ed6-b4f7-dc7e-cb92aa521c9f.htm). Is it missing the [**OAuth2Client**](https://github.com/intuit/QuickBooks-V3-DotNET-SDK/blob/master/IPPDotNetDevKitCSV3/Code/Intuit.Ipp.OAuth2PlatformClient/Client/OAuth2Client.cs) class or am I going crazy?

**Comment by nimisha84** [comment]
It is available here-https://developer.intuit.com/app/developer/qbo/docs/develop/authentication-and-authorization/oauth-2.0

**Comment by pideltajah** [comment]
> It is available here-https://developer.intuit.com/app/developer/qbo/docs/develop/authentication-and-authorization/oauth-2.0

That is an OAuth 2.0 tutorial, not [OAuth2Client](https://github.com/intuit/QuickBooks-V3-DotNET-SDK/blob/master/IPPDotNetDevKitCSV3/Code/Intuit.Ipp.OAuth2PlatformClient/Client/OAuth2Client.cs) class API documentation.

**Comment by pideltajah** [comment]
Why was this closed without being resolved?

**Comment by pideltajah** [comment]
[OAuth2PlatformClient.zip](https://github.com/intuit/QuickBooks-V3-DotNET-SDK/files/6086591/OAuth2PlatformClient.zip)
I think that file contains the documentation XML file. It has several sections like `<member name="T:Intuit.Ipp.OAuth2PlatformClient.OAuth2Client">`. It doesn't look like great documentation but I believe it was generated from the [source code's](https://github.com/intuit/QuickBooks-V3-DotNET-SDK) xml doc comments. I tried to use [DocFX](https://dotnet.github.io/docfx/tutorial/docfx_getting_started.html#3-use-docfx-integrated-with-visual-studio) to generate an html out of it but I gave up.

**Comment by nimisha84** [comment]
Please check out the https://developer.intuit.com/app/developer/qbo/docs/develop/authentication-and-authorization/oauth-2.0 -> there is code for each language and all code related to OAuth2Client is available. What exactly are you looking for in the docs wrt to SDK? The oAuth2 docs explain the concept and the SDK examples show how that can be done using OAuth2PlatformClient. Let me know if that helps

**Comment by pideltajah** [comment]
@nimisha84 Thanks for your time, it is appreciated. I know the code is available, I had to go through it to figure out the OAuth2Client class when implementing QBO OAuth 2 authentication  in my app. My point is that it is not documented, it is omitted from the documentation of the OAuth2PlatformClient namespace. The link you keep posting is not API documentation. Yes, the code is available, which is great, but it would be much better if we could just get the documentation for that class added to all the other documentation ([https://static.developer.intuit.com/sdkdocs/qbv3doc/ippdotnetdevkitv3/html/ccf8123c-8ed6-b4f7-dc7e-cb92aa521c9f.htm](https://static.developer.intuit.com/sdkdocs/qbv3doc/ippdotnetdevkitv3/html/ccf8123c-8ed6-b4f7-dc7e-cb92aa521c9f.htm)). I don't see why you would not want this done. I have very little experience with .NET technology or I might be of more help generating the documentation from the XML comments.

TL;DR:

- The documentation is missing.
- The tutorial is not a substitute for API documentation.
- It should be simple enough for someone who knows what they are doing to add the missing documentation.

**Comment by nimisha84** [comment]
Agreed, this static doc has not been updated in a long time as we did took a survey from previous devs if they read it at any point of time and we got a very minimal response and so, if you see those static docs for all SDKs haven't been updated in a while. We will be revamping the docs again soon and will again the take the input from other devs too to decide if we do want to host them again. The link you found is a really old one but hasn't been taken down. Thanks for your feedback. I think if you had found the link I shared for OAuth earlier, that would have solved for the use case you had but unfortunately you had to go through the trouble of figuring it out. We are also trying to solve for the organization of docs so that it is easier to find out the info you are looking for in future.
